stub/src/main/java/io/grpc/kotlin/CoroutineContextServerInterceptor.kt

```diff
@@ -49,8 +53,15 @@ abstract class CoroutineContextServerInterceptor : ServerInterceptor {
     call: ServerCall<ReqT, RespT>,
     headers: Metadata,
     next: ServerCallHandler<ReqT, RespT>
-  ): ServerCall.Listener<ReqT> =
-    withGrpcContext(GrpcContext.current().extendCoroutineContext(coroutineContext(call, headers))) {
+  ): ServerCall.Listener<ReqT> {
+    val coroutineContext = try {
+      coroutineContext(call, headers)
+    } catch (e: StatusException) {
+      call.close(e.status, e.trailers ?: Metadata())
+      return object: ServerCall.Listener<ReqT>() {}
```

Comment:
Can we make this listener actually throw instead of just no-op?  Failing "loudly" by throwing an exception makes it clearer what's going on than simple a no-op listener.

Reply:
I've modified it to rethrow the exception, though that doesn't appear to affect observable behaviour, so I am not sure how to test that change. But if it's just about clarity, then this should suffice.